packages/middleware-endpoint/src/getEndpointPlugin.ts

```diff
@@ -1,24 +1,25 @@
 import { Pluggable, SerializeHandlerOptions } from "@aws-sdk/types";
 
 import { endpointMiddleware } from "./endpointMiddleware";
-import { EndpointParameterInstruction } from "./types";
+import { EndpointResolvedConfig } from "./resolveEndpointConfig";
+import { EndpointParameterInstructions } from "./types";
 
 export const endpointMiddlewareOptions: SerializeHandlerOptions = {
   step: "serialize",
   tags: ["ENDPOINT_PARAMETERS", "ENDPOINT_V2", "ENDPOINT"],
-  name: "endpointMiddleware",
+  name: "endpointsV2Middleware",
```

Comment:
nit: singular endpointV2, like we've used in other places
```suggestion
  name: "endpointV2Middleware",
```